app/src/main/java/com/okta/oidc/example/MyConnectionFactory.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2019, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License,
+ * Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the
+ * License.
+ */
+
+package com.okta.oidc.example;
+
+import androidx.annotation.NonNull;
+
+import com.okta.oidc.net.HttpConnectionFactory;
+import com.okta.oidc.net.OktaHttpClient;
+
+/**
+ * Simple connection factory.
+ */
+public class MyConnectionFactory implements HttpConnectionFactory {
```

Comment:
This factory looks redundant as long as we do not have any parameters for our OktaHttpClients.

Reply:
It is mainly for testing. In the unit tests it is passing in the type of client in the method setClientType. Or do you mean we don't need a HttpConnectionFactory at all and just pass in a HttpClient in the builder?

Reply:
I would say that it is better provide HttpClient directly into builder because:
- It would be easier for end user to add his own implementation of the client (they will not need to modify our factory)
- As long as we have only empty constructor, I don't think we need a factory.

Reply:
yes, agreed. Will remove.